backend/src/main/java/de/bund/digitalservice/ris/norms/adapter/output/database/service/VerkuedungImportProcessDBService.java

```diff
@@ -34,4 +42,70 @@ public Optional<VerkuendungImportProcess> loadVerkuendungImportProcess(
       .findById(command.id())
       .map(VerkuendungImportProcessMapper::mapToDomain);
   }
+
+  @Override
+  public VerkuendungImportProcess saveOrUpdateVerkuendungImportProcess(
+    SaveVerkuendungImportProcessPort.Command command
+  ) {
+    final Optional<VerkuendungImportProcessDto> optionalExistingVerkuendungImportProcessDto =
+      verkuendungImportProcessesRepository.findById(command.id());
+
+    if (optionalExistingVerkuendungImportProcessDto.isPresent()) {
+      var existingVerkuendungImportProcessDto = optionalExistingVerkuendungImportProcessDto.get();
+      updateProcessDto(command, existingVerkuendungImportProcessDto);
+      return VerkuendungImportProcessMapper.mapToDomain(existingVerkuendungImportProcessDto);
+    } else {
+      VerkuendungImportProcessDto createdVerkuendungImportProcessDto =
+        new VerkuendungImportProcessDto();
+      createdVerkuendungImportProcessDto.setId(command.id());
+      updateProcessDto(command, createdVerkuendungImportProcessDto);
+      return VerkuendungImportProcessMapper.mapToDomain(
+        verkuendungImportProcessesRepository.save(createdVerkuendungImportProcessDto)
+      );
+    }
+  }
+
+  private static void updateProcessDto(
+    SaveVerkuendungImportProcessPort.Command command,
+    VerkuendungImportProcessDto verkuendungImportProcessDto
+  ) {
+    final VerkuendungImportProcessDto.Status status = VerkuendungImportProcessMapper.mapStatusToDto(
+      command.status()
+    );
+    verkuendungImportProcessDto.setStatus(status);
+
+    switch (status) {
+      case CREATED -> verkuendungImportProcessDto.setCreatedAt(Instant.now());
+      case PROCESSING -> verkuendungImportProcessDto.setStartedAt(Instant.now());
+      case SUCCESS -> verkuendungImportProcessDto.setFinishedAt(Instant.now());
+      case ERROR -> {
+        verkuendungImportProcessDto.setFinishedAt(Instant.now());
+
+        final List<VerkuendungImportProcessDetailDto> incomingDetails = command
+          .details()
+          .stream()
+          .map(VerkuendungImportProcessMapper::mapDetailToDto)
+          .toList();
+
+        final Set<String> existingDetailSet = verkuendungImportProcessDto
+          .getDetail()
+          .stream()
+          .map(VerkuendungImportProcessDetailDto::getDetail)
+          .collect(Collectors.toSet());
+
+        for (VerkuendungImportProcessDetailDto incomingDetail : incomingDetails) {
+          if (!existingDetailSet.contains(incomingDetail.getDetail())) {
+            final VerkuendungImportProcessDetailDto safeDetail = VerkuendungImportProcessDetailDto
+              .builder()
+              .type(incomingDetail.getType())
+              .title(incomingDetail.getTitle())
+              .detail(incomingDetail.getDetail())
+              .build();
+            verkuendungImportProcessDto.getDetail().add(safeDetail);
+          }
+        }
+      }
+      default -> log.error("Unhandled status {}", status);
```

Comment:
i dont think we need a default case here and with it missing the compiler catches it if we add a new status